ttnn/cpp/ttnn/operations/pool/upsample/upsample.cpp

```diff
@@ -31,21 +77,12 @@ static MemoryConfig compute_bilinear_autoshard_memory_config(const ttnn::Tensor&
     const uint32_t max_num_cores = compute_grid_size.x * compute_grid_size.y;
     const uint32_t max_num_shards = batch_size * input_h * input_w;
     uint32_t num_shards = std::min(max_num_cores, max_num_shards);
-    while (num_shards > 0) {
-        /*
-        Operation currently requires that each core takes an equal number of input sticks,
-        and that the number of sticks per core is divisible by width, ie every core takes
-        whole rows as input. The below condition ensures that.
-        Maximizing the number of shards under said condition.
-        */
-        if (batch_size * input_h % num_shards == 0) {
-            break;
-        }
-        num_shards--;
-    }
+    // Calculate input shard with padding (like pool2d)
```

Comment:
let's simplify this code and remove const we don't eed as memory_layout, buffer_type...